contracts/SafeRelayBoundedRefund.sol

```diff
@@ -152,20 +155,27 @@ contract SafeRelayBoundedRefund is BoundaryManager, ReentrancyGuard {
      * @return payment Amount of refunded gas
      */
     function handleRefund(RefundParams calldata refundParams, uint256 startGas) internal returns (uint256 payment) {
+        // COMMENT: maybe good to extract the address(0) to a named constant. 
+
         // solhint-disable-next-line avoid-tx-origin
         address payable receiver = refundParams.refundReceiver == address(0) ? payable(tx.origin) : refundParams.refundReceiver;
 
         uint256 gasConsumed = startGas - gasleft() + COVERED_REFUND_PAYMENT_GAS;
         payment = min(gasConsumed, refundParams.gasLimit) * refundParams.maxFeePerGas;
 
         if (refundParams.gasToken == NATIVE_TOKEN) {
+            // COMMENT: for defensive programming, I would extract the output of the `execute` into a separate variable and 
+            // check the variable in the if statement.
             if (!execute(refundParams.safeAddress, receiver, payment, "0x")) {
                 revert RefundFailure();
             }
         } else {
             // 0xa9059cbb - keccack("transfer(address,uint256)")
             bytes memory data = abi.encodeWithSelector(0xa9059cbb, receiver, payment);
             if (!execute(refundParams.safeAddress, refundParams.gasToken, 0, data)) {
+                // COMMENT: I would add some text or a different error type to distinguish between
+                // native refund failure and token refund failure.
+                // Also, if there's a way to get more information why token transfer failed, that would help
```

Comment:
I'm unsure if the second is possible (information about why it failed) because some tokens return false to indicate a failed transfer and some revert. Would be keen to hear what @rmeissner thinks

Reply:
I would not depend to much on a token implementation. To be honest I would have removed the option to pay with a token and use a different module for that :P

Reply:
>To be honest I would have removed the option to pay with a token and use a different module for that :P
Less reasons to change this contract in the future if the token interface or usage changes

And if to remove paying with tokens completely, then it would be less complicated.

Per my comment, I thought having distinct error messages would ease the support downstream when people would see that their transaction failed, then it would be easier to pinpoint the reason.

Reply:
Let's wait for @sche; I think users want to pay gas with stable coins. And IMO, it's pretty unlikely that a new token interface comes up without ERC20 compatibility

Reply:
I very much agree with Dima here (and commented this on another point). Trying to solve all at once and implement all logic is not always the best way (especially to test something).

I did this in the Safe contract and in the Allowance module and now there is unused code, which is more a liability then of use. As modules are in general very easy to add to a Safe I would recommend to focus a module on a specific use case and rather have 2 modules (1 for token payment and 1 for native coin payment). This reduces complexity and makes it easier to review/ audit the contract and therefore minimizes risk.